src/plugin/ipc/file/ptyconnlist.cpp

```diff
@@ -61,6 +63,7 @@ PtyConnList::instance()
 void
 PtyConnList::drain()
 {
+  virtPtyId = SharedData::getVirtualPtyId();
```

Comment:
Do all processes on a same node return the same `virtPtyId`? Also, could you rename it to something like `nextVirtPtyId`, if that is what you intended? And the same for the function names.

Reply:
The processes request for a new virtual pty id through existing `SharedData` API. The problem is that we don't save and restore the last virt pty id that was in-use at checkpoint time, and so on restart, we end up reallocating in-use virt pty ids for any new ptys.

The intent here is *not* to get a new virt pty id; rather, it is to save and restore the last used
virt pty id. This is to ensure that when a process requests for a new pty post restart, it must not reuse any of the pre-exiting, in-use ids.